### DISCUSSION TITLE
How Can I change WebGL information?

### DISCUSSION THREAD

**Discussion by damidoug** [seleniumbase/SeleniumBase#2248] [Q&A] [answered]
When we test on https://pixelscan.net/, the WebGL information matches that of my computer. Is there a way to insert random information? It's as if we could introduce a method to randomize hardware details, the operating system (Windows, Linux, Mac, etc.), screen resolution, and more.

**Comment by mdmintz** [comment, marked as answer]
Yes, that's been done before. Even before UC Mode existed, there was an integration with `selenium-stealth` (https://github.com/seleniumbase/SeleniumBase/issues/942#issuecomment-880771150), which lets you "change" (fake) various properties:

You can integrate it directly into the custom ``setUp()`` method of your tests like this after you pip install it:

```python
from seleniumbase import BaseCase
from selenium_stealth import stealth

class BaseTestCase(BaseCase):
    def setUp(self):
        super().setUp()
        stealth(self.driver,
            languages=["en-US", "en"],
            vendor="Google Inc.",
            platform="Win32",
            webgl_vendor="Intel Inc.",
            renderer="Intel Iris OpenGL Engine",
            fix_hairline=True,
        )
```

Then have your test classes inherit `BaseTestCase` instead of `BaseCase`. (See [SeleniumBase/help_docs/syntax_formats.md#sb_sf_02](https://github.com/seleniumbase/SeleniumBase/blob/master/help_docs/syntax_formats.md#sb_sf_02))

There's general information on StackOverflow about that here: https://stackoverflow.com/q/70265306/7058266

Additionally, you could use `driver.execute_cdp_cmd()` to make internal modifications to your Chrome settings.
You would need to be familiar with CDP internals: https://chromedevtools.github.io/devtools-protocol/tot/Network/.
For example, here's a SeleniumBase script that changes your GeoLocation: [SeleniumBase/examples/test_geolocation.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/test_geolocation.py)

**Reply by dimaslanjaka** [reply]
I tried your codes, but error. WebGL never queried/empty.

![image](https://github.com/user-attachments/assets/f0b918e0-d8fc-48d3-9f97-67c79686e95d)

**Reply by mdmintz** [reply]
Try the newer example: https://github.com/seleniumbase/SeleniumBase/blob/master/examples/cdp_mode/raw_pixelscan.py
